### ISSUE TITLE
Deterministic bootstrapping

### ISSUE THREAD

**Issue by taktoa** [hercules-ci/hercules#51]
In [pkgs.nix](https://github.com/hercules-ci/hercules/blob/master/pkgs.nix) you can use the deterministic bootstrapping style described in https://github.com/NixOS/nix/issues/1299#issuecomment-316842769 to prevent the user from having to redownload the tarball hourly.

**Comment by aneeshusa** [comment]
I also like the trick from https://github.com/NixOS/nix/issues/1528, which seems a little nicer to me.

**Comment by taktoa** [comment]
sure, though that requires a git submodule containing the relevant nixpkgs revision

**Comment by domenkozar** [comment]
Until Nix 1.12 is released I guess it's best to use https://github.com/input-output-hk/iohk-ops/blob/staging/fetch-nixpkgs.nix

**Comment by taktoa** [comment]
@domenkozar I find it likely that that file was written after I conversation I had with @cleverca22 this summer (i.e.: it's probably equivalent to [the gist in that comment](https://gist.github.com/taktoa/4ec8b0cb6d07857400b07a830ad99558))

**Comment by domenkozar** [comment]
Yeah, it goes around also by @zimbatm at https://github.com/numtide/todomvc-nix/commit/f3c7c35b00dc0c5897a08531e5902e09110aeb1a
